### ISSUE TITLE
Hadolint error when heredoc delimiter is escaped with \ instead of quoting

### ISSUE THREAD

**Issue by gwerbin-tive** [hadolint/hadolint#1051]
- [x] This is a bug report
- [ ] This is a feature request
- [x] I searched existing issues before opening this one

### Expected behavior

No error

### Actual behavior

```
tmp.dockerfile:4:3 unexpected 'e' expecting '#', '\', ADD, ARG, CMD, COPY, ENTRYPOINT, ENV, EXPOSE, FROM, HEALTHCHECK, LABEL, MAINTAINER, ONBUILD, RUN, SHELL, STOPSIGNAL, USER, VOLUME, WORKDIR, a pragma, at least one space, or end of input
```

### Steps to reproduce the behavior

Hadolint v2.12.0

```bash
hadolint tmp.dockerfile
```

Causes error:

```Dockerfile
FROM debian

COPY <<-\EOT /data.txt
  echo '$TARGETBUILDARCH'
EOT

CMD ["cat", "/data.txt"]
```

Works fine:

```Dockerfile
FROM debian

COPY <<-"EOT" /data.txt
  echo '$TARGETBUILDARCH'
EOT

CMD ["cat", "/data.txt"]
```
